### ISSUE TITLE
Unable to install Python version on MacOS, solved...

### ISSUE THREAD

**Issue by georgalis** [pyenv/pyenv#1352]
I have very similar problems to #1348 but mine is solved. This may or not be a solution for #1348 In my case, the issue is the same anticipated by apple and described here:

> https://developer.apple.com/documentation/xcode_release_notes/xcode_10_release_notes#3035624
> The command line tools will search the SDK for system headers by default. However, some software may fail to build correctly against the SDK and require macOS headers to be installed in the base system under /usr/include. If you are the maintainer of such software, we encourage you to update your project to work with the SDK or file a bug report for issues that are preventing you from doing so. As a workaround, an extra package is provided which will install the headers to the base system. In a future release, this package will no longer be provided. You can find this package at:
> 
> /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg
> 

```
# find /Library/Developer/CommandLineTools/ -name \*pkg
{null}
# xcode-select -v
xcode-select version 2354.
```
Although I have commanlinetools installed and selected the libraries are not found...

```
# pyenv install 3.7.3
python-build: use openssl@1.1 from homebrew
python-build: use readline from homebrew
Downloading Python-3.7.3.tar.xz...
-> https://www.python.org/ftp/python/3.7.3/Python-3.7.3.tar.xz
Installing Python-3.7.3...
python-build: use readline from homebrew

BUILD FAILED (OS X 10.14.5 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/j5/kkpg6pp96f9fnqdqs3_gqr7w0000gn/T/python-build.20190608163159.45426
Results logged to /var/folders/j5/kkpg6pp96f9fnqdqs3_gqr7w0000gn/T/python-build.20190608163159.45426.log

Last 10 log lines:
  File "/private/var/folders/j5/kkpg6pp96f9fnqdqs3_gqr7w0000gn/T/python-build.20190608163159.45426/Python-3.7.3/Lib/ensurepip/__main__.py", line 5, in <module>
    sys.exit(ensurepip._main())
  File "/private/var/folders/j5/kkpg6pp96f9fnqdqs3_gqr7w0000gn/T/python-build.20190608163159.45426/Python-3.7.3/Lib/ensurepip/__init__.py", line 204, in _main
    default_pip=args.default_pip,
  File "/private/var/folders/j5/kkpg6pp96f9fnqdqs3_gqr7w0000gn/T/python-build.20190608163159.45426/Python-3.7.3/Lib/ensurepip/__init__.py", line 117, in _bootstrap
    return _run_pip(args + [p[0] for p in _PROJECTS], additional_paths)
  File "/private/var/folders/j5/kkpg6pp96f9fnqdqs3_gqr7w0000gn/T/python-build.20190608163159.45426/Python-3.7.3/Lib/ensurepip/__init__.py", line 27, in _run_pip
    import pip._internal
zipimport.ZipImportError: can't decompress data; zlib not available
make: *** [install] Error 1

```
I was able to successfully install by adding the CFLAGS env per https://github.com/pyenv/pyenv/wiki/Common-build-problems

```
CFLAGS="-I$(xcrun --show-sdk-path)/usr/include" pyenv install 3.7.3
python-build: use openssl@1.1 from homebrew
python-build: use readline from homebrew
Downloading Python-3.7.3.tar.xz...
-> https://www.python.org/ftp/python/3.7.3/Python-3.7.3.tar.xz
Installing Python-3.7.3...
python-build: use readline from homebrew
Installed Python-3.7.3 to /Users/geo/.pyenv/versions/3.7.3
```

Thanks for providing the common build problems wiki, is it possible to incorporate the `xcrun --show-sdk-path` command into `pyenv install` command sequence?

**Comment by georgalis** [comment]
Added this functionality to `pyenv-install` but I don't have permission for a pull request...
```
# git diff master 20190608-search-sdk-headers
diff --git a/plugins/python-build/bin/pyenv-install b/plugins/python-build/bin/pyenv-install
index 22333f6e..1890cc39 100755
--- a/plugins/python-build/bin/pyenv-install
+++ b/plugins/python-build/bin/pyenv-install
@@ -51,0 +52,8 @@ fi
+# Library path help
+case $(uname) in
+  Darwin)
+    # if unset, assign default
+    [ "$CFLAGS" ] || export CFLAGS="-I$(xcrun --show-sdk-path)/usr/include"
+  ;;
+esac
+
```

**Comment by joshfriend** [comment]
The `xcrun --show-sdk-path` solution is already well documented in other issues. I have a fix in progress in #1339

> I don't have permission for a pull request...

Huh?

**Comment by georgalis** [comment]
On Sun, Jun 9, 2019 at 9:23 AM Josh Friend <notifications@github.com> wrote:

> The xcrun --show-sdk-path solution is already well documented in other
> issues. I have a fix in progress in #1339
> <https://github.com/pyenv/pyenv/pull/1339>
>

Great! I see this goes back to Sept 2018. I set aside Sat to learn pyenv,
but all my time got into patching an issue that already seems fixed.

Why not release a fix?

> I don't have permission for a pull request...
>
> Huh?
>
Glad that you are interested in doing everything right, but forking seems a
bit heavy for my simple patch.

Is your priority getting logic for every odd case worked out and CI tests?
I vote for a patch that solves most of the problems most of the time, then
if it takes 9 months to improve, most people won't even notice.

-George


-- 
George Georgalis, (415) 894-2710, http://www.galis.org/

**Comment by gennaro-tedesco** [comment]
I am seeing the same problem: what is the recommended solution (since this is marked as solved)?
